src/chrdev.rs

```diff
@@ -11,29 +10,30 @@ use crate::c_types;
 use crate::error::{Error, KernelResult};
 use crate::user_ptr::{UserSlicePtr, UserSlicePtrWriter};
 
-pub fn builder(name: &'static str, minors: Range<u16>) -> KernelResult<Builder> {
+pub fn builder(name: &'static str) -> KernelResult<Builder> {
     if !name.ends_with('\x00') {
         return Err(Error::EINVAL);
     }
 
     return Ok(Builder {
         name,
-        minors,
+        0,
         file_ops: vec![],
     });
 }
 
 pub struct Builder {
     name: &'static str,
-    minors: Range<u16>,
+    minor_start: u16,
     file_ops: Vec<&'static FileOperationsVtable>,
 }
 
 impl Builder {
+    pub fn start_minor_at(&mut self, minor_start: u16) {
```

Comment:
I think this should take `self` and return `Buider`, like `register_device` does.